cli/src/main/kotlin/tools/samt/cli/CliCompiler.kt

```diff
@@ -3,45 +3,52 @@ package tools.samt.cli
 import tools.samt.common.DiagnosticConsole
 import tools.samt.common.DiagnosticContext
 import tools.samt.lexer.Lexer
+import tools.samt.parser.FileNode
 import tools.samt.parser.Parser
 import tools.samt.parser.ParserException
 import java.io.File
 
-fun parse(files: List<String>) {
-    for (it in files) {
-        val file = File(it)
+fun parse(filePaths: List<String>): List<FileNode> = buildList {
+    for (filePath in filePaths) {
+        val file = File(filePath)
         if (!file.exists()) {
-            println("File '$it' does not exist")
+            println("File '$filePath' does not exist")
             continue
         }
         if (!file.canRead()) {
-            println("File '$it' cannot be read, bad file permissions?")
+            println("File '$filePath' cannot be read, bad file permissions?")
             continue
         }
         if (file.extension != "samt") {
-            println("File '$it' must end in .samt")
+            println("File '$filePath' must end in .samt")
             continue
         }
 
         val source = file.readText()
         val diagnostics = DiagnosticConsole(DiagnosticContext(file.canonicalPath, source))
 
-        val tokenStream = Lexer.scan(source.reader(), diagnostics)
+        var parsedFileNode = parseSource(file.canonicalPath, source, diagnostics)
 
         if (diagnostics.hasErrors()) {
             diagnostics.printToConsole()
             continue
         }
 
-        val fileNode = try {
-            Parser.parse(tokenStream, diagnostics)
-        } catch (e: ParserException) {
-            diagnostics.printToConsole()
-            continue
-        }
+        add(parsedFileNode!!)
+    }
+}
+
+fun parseSource(filePath: String, fileSource: String, diagnostics: DiagnosticConsole): FileNode? {
+    val tokenStream = Lexer.scan(fileSource.reader(), diagnostics)
+
+    if (diagnostics.hasErrors()) {
+        return null
+    }
 
-        diagnostics.printToConsole()
-        println(fileNode)
+    return try {
+        Parser.parse(filePath, tokenStream, diagnostics)
+    } catch (e: ParserException) {
+        null
```

Comment:
Maybe add a comment like `The parse exception message in also added to diagnostics, we can safely ignore it here`